ethereum/aave/insert_aave_tokens.sql

```diff
@@ -0,0 +1,169 @@
+CREATE OR REPLACE FUNCTION aave.insert_aave_tokens(start_time timestamptz, end_time timestamptz) RETURNS integer
+LANGUAGE plpgsql AS $function$
+DECLARE r integer;
+BEGIN
+WITH rows AS (
+    INSERT INTO aave.aave_tokens (
+      token_address,
+      decimals,
+      symbol,
+      underlying_token_address,
+      underlying_token_symbol,
+      side,
+	token_name,
+	program_type
+    )
+
+WITH standard_tokens AS (
+SELECT
+a."aToken" AS token_address,
+e.decimals AS decimals,
+'a' || e.symbol AS symbol,
+a.asset AS underlying_token_address,
+e.symbol AS underlying_token_symbol,
+'Deposit' AS side
+FROM aave_v2."LendingPoolConfigurator_evt_ReserveInitialized" a --asset is raw, atoken is atoken
+
+LEFT JOIN erc20."tokens" e
+ON a.asset = e."contract_address"
+
+UNION --ammtokens
+
+SELECT 
+    amm."aTokenAddress" AS token_address,
+    e.decimals AS decimals,
+    'aAmm' || e.symbol AS symbol,
+    amm.asset AS underlying_token_address,
+    e.symbol AS underlying_token_symbol,
+    'AMM' AS side
+    FROM aave_v2."AaveLendingPoolV2AMM_call_initReserve" amm
+    LEFT JOIN erc20."tokens" e
+        ON amm.asset = e."contract_address"
+    WHERE e.decimals IS NOT NULL
+    AND call_block_time >= start_time
+    AND call_block_time < end_time
+
+UNION --variable debt
+
+SELECT 
+    vd."variableDebtToken" AS token_address,
+    e.decimals AS decimals,
+    'variableDebt' || e.symbol AS symbol,
+    vd.asset AS underlying_token_address,
+    e.symbol AS underlying_token_symbol,
+    'Borrow' AS side
+    FROM aave_v2."LendingPoolConfigurator_evt_ReserveInitialized" vd
+    LEFT JOIN erc20."tokens" e
+        ON vd.asset = e."contract_address"
+    WHERE e.decimals IS NOT NULL
+    AND evt_block_time >= start_time
+    AND evt_block_time < end_time
+
+UNION --stable debt 
+
+SELECT 
+    sd."stableDebtToken" AS token_address,
+    e.decimals AS decimals,
+    'stableDebt' || e.symbol AS symbol,
+    sd.asset AS underlying_token_address,
+    e.symbol AS underlying_token_symbol,
+    'Borrow' AS side
+    FROM aave_v2."LendingPoolConfigurator_evt_ReserveInitialized" sd
+    LEFT JOIN erc20."tokens" e
+        ON sd.asset = e."contract_address"
+    WHERE e.decimals IS NOT NULL
+    AND evt_block_time >= start_time
+    AND evt_block_time < end_time
+)
+
+, additional_tokens AS ( --i.e. ARC, RWA, etc.
+-- deposit tokens
+SELECT 
+l.contract_address AS token_address,
+bytea2numeric(substring(data from 65 for 32)) AS decimals,
+regexp_replace(REPLACE(encode(substring(data from 288 for 128),'escape')::text,'\000',''), '\W', ' ', 'g') AS symbol,
+dune_user_generated.get_address_from_data(topic2) AS underlying_token_address,
```

Comment:
This should be added as a abstraction function, so it's not deleted.

Reply:
Added the function in public to this PR